packages/manager/apps/identity-access-management/src/pages/tagManager/components/tagsListDatagrid/utils/sortTags.ts

```diff
@@ -8,7 +8,19 @@ export type SortTagsParams = {
 
 export const sortTags = ({ tags, columnId, desc }: SortTagsParams) => {
   return tags.sort((tagA, tagB) => {
-    if (desc) return tagA[columnId] < tagB[columnId] ? 1 : -1;
-    return tagA[columnId] > tagB[columnId] ? 1 : -1;
+    if (typeof tagA[columnId] === 'string') {
+      if (desc)
+        return tagB[columnId]
+          .toString()
+          .localeCompare(tagA[columnId].toString());
+      return tagA[columnId].toString().localeCompare(tagB[columnId].toString());
+    }
+
+    if (typeof tagA[columnId] === 'number') {
+      if (desc) return Number(tagB[columnId]) - Number(tagA[columnId]);
+      return Number(tagA[columnId]) - Number(tagB[columnId]);
+    }
+
+    return 0;
   });
```

Comment:
Just as suggestion because I known this code exists on app

To not repeat yourself you can :


```suggestion
export const sortTags = ({ tags, columnId, desc }: SortTagsParams) => {
return tags.sort((tagA, tagB) => {
    if (typeof tagA[columnId] === 'string') {
      if (desc)
        return sortString(tagB[columnId], tagA[columnId])
      return sortString(tagA[columnId] tagB[columnId]
    }

    if (typeof tagA[columnId] === 'number') {
      if (desc) return sortNumber(tagB[columnId], tagA[columnId]
      return sortNumber(tagA[columnId], tabB[columnId])
    }

    return 0;
});
```

With that is more testable, more composable, and the single responsability of your function it's to call the good function from the context (Type and sortype)